File: yh_difference/src/yh_difference_server.cpp
#include "ros/ros.h"
#include "yh_difference/YhDifference.h"

bool difference(yh_difference::YhDifference::Request& req, yh_difference::YhDifference::Response& res)
{
    if (req.a > req.b)
    {
        res.result = req.a - req.b;
    }
    else
    {
        res.result = req.b - req.a;
    }

    return true;
}

int main(int argc, char** argv)
{
    ros::init(argc, argv, "yh_difference_server");
    ros::NodeHandle nh;

    ros::ServiceServer server = nh.advertiseService("yh_difference_service", difference);

    ros::spin();

    return 0;
}

File: yh_difference/scripts/yh_difference_server.py
#!/usr/bin/python3
# -*- coding: utf-8 -*-
import rospy
from yh_difference.srv import YhDifference, YhDifferenceResponse

def difference(req):
    res = YhDifferenceResponse()
    res.result = abs(req.a - req.b)
    return res

def difference_server():
    rospy.init_node("yh_difference_server")
    rospy.Service("yh_difference_service", YhDifference, difference)
    rospy.spin()

if __name__ == "__main__":
    difference_server()